server/src/main/java/org/elasticsearch/index/shard/ShardPath.java

```diff
@@ -192,85 +189,22 @@ public static ShardPath selectNewPathForShard(NodeEnvironment env, ShardId shard
 
         if (indexSettings.hasCustomDataPath()) {
             dataPath = env.resolveCustomLocation(indexSettings.customDataPath(), shardId);
-            statePath = env.nodePaths()[0].resolve(shardId);
+            statePath = env.nodePath().resolve(shardId);
         } else {
             BigInteger totFreeSpace = BigInteger.ZERO;
```

Comment:
I think the whole `else` block can be replaced by this one line:

```
            dataPath = statePath = env.nodePath().resolve(shardId);
```

The comments etc. are not important, nor is the checking of the free space.